clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -80,54 +80,33 @@ class GCEUtil {
                           GoogleExecutorTraits executor) {
     task.updateStatus phase, "Looking up image $imageName..."
 
+    def filter = "name eq $imageName"
     def imageProjects = [projectName] + credentials?.imageProjects + baseImageProjects - null
     def sourceImage = null
-    // We want predictable iteration order that matches the order of insertion.
-    def imageProjectToNextPageTokenMap = new LinkedHashMap<String, String>()
 
-    // This will ensure that each image project is queried.
-    imageProjects.each { imageProjectToNextPageTokenMap[it] = null }
-
-    while (!sourceImage && imageProjectToNextPageTokenMap) {
-      def imageListBatch = buildBatchRequest(compute, clouddriverUserAgentApplicationName)
-      def imageListCallback = new JsonBatchCallback<ImageList>() {
-        @Override
-        void onFailure(GoogleJsonError e, HttpHeaders responseHeaders) throws IOException {
-          updateStatusAndThrowNotFoundException("Error locating $imageName in these projects: $imageProjects: $e.message", task, phase)
-        }
-
-        @Override
-        void onSuccess(ImageList imageList, HttpHeaders responseHeaders) throws IOException {
-          // No need to look through these images if the requested image was already found.
-          if (!sourceImage) {
-            sourceImage = imageList.items.find { it.name == imageName }
-          }
-
-          // selfLinks look like this: https://www.googleapis.com/compute/alpha/projects/ubuntu-os-cloud/global/images
-          def selfLinkTokens = imageList.getSelfLink().tokenize("/")
-          def imageProject = selfLinkTokens[selfLinkTokens.size() - 3]
-
-          if (imageList.nextPageToken) {
-            imageProjectToNextPageTokenMap[imageProject] = imageList.nextPageToken
-          } else {
-            imageProjectToNextPageTokenMap.remove(imageProject)
-          }
-        }
+    def imageListBatch = buildBatchRequest(compute, clouddriverUserAgentApplicationName)
+    def imageListCallback = new JsonBatchCallback<ImageList>() {
+      @Override
+      void onFailure(GoogleJsonError e, HttpHeaders responseHeaders) throws IOException {
+        updateStatusAndThrowNotFoundException("Error locating $imageName in these projects: $imageProjects: $e.message", task, phase)
```

Comment:
404 won't ever be caught here, even if the image does not exist (true, throwing 'not found' doesn't seem perfect here).

What I want to avoid is getting the wrong image due to something like a network error. Just logging all of the errors wouldn't be right, but throwing on all of them is also not right either.

Dan also raised the idea of not throwing the error and just logging it. The order of precedence does take into account the order of the image projects. First, the managed project, then any image projects, and lastly the public base image projects.

Would you be ok with this going through as-is, since it's an optimization of the existing logic, and then I'll open a new bug to apply retry logic here and give more thought to the error-handling?